### PR TITLE
planning: planning_path_debug branch merge back to master

### PR DESCRIPTION
**Pull request by YajiaZhang**
planning_path_debug branch merge back to master with new feature switched off.

### CODE REVIEW

File: modules/planning/conf/planning.conf [hunk 1]
```diff
@@ -14,6 +14,8 @@
 --default_cruise_speed=20.00
 --enable_scenario_side_pass
 --enable_scenario_traffic_light
+--enable_collision_detection
```
Comment (Capri2014) [inline modules/planning/conf/planning.conf]: @YajiaZhang This flag has been deprecated via commit https://github.com/ApolloAuto/apollo/commit/1eff2859c13803037d62599cb14415a0f6b4f066  Please change accordingly.

Comment (YajiaZhang) [reply modules/planning/conf/planning.conf]: Done.

File: modules/planning/math/piecewise_jerk/BUILD [hunk 2]
```diff
@@ -3,12 +3,12 @@ load("//tools:cpplint.bzl", "cpplint")
 package(default_visibility = ["//visibility:public"])
 
 cc_library(
-    name = "fem_1d_qp_problem",
+    name = "piecewise_jerk_problem",
```
Comment (Capri2014) [inline modules/planning/math/piecewise_jerk/BUILD]: If we decided to fully deprecate "fem_1d_qp_problem" files, a better way is to remove the corresponding source files as well. --- If the depreciation is depending on the test result and just to change the default as "piecewise_jerk_problem", I would recommend having them both in compile and test and just use a flag to switch between which one you use. --- Either way works, but if we only remove the "piecewise_jerk_problem" from BUILD but not their source files as well. But we may accidentally leave dead code of fem_1d_qp_problem.cc/h without protection by compilation or test.

File: modules/planning/math/piecewise_jerk/BUILD [hunk 3]
```diff
@@ -18,16 +18,28 @@ cc_library(
...
     ],
     deps = [
-        ":fem_1d_qp_problem",
+        ":piecewise_jerk_problem",
         "//cyber/common:log",
         "@gtest//:main",
     ],
 )
 
+cc_library(
```
Comment (Capri2014) [inline modules/planning/math/piecewise_jerk/BUILD]: Was the test of "fem_1d_qp_problem" removed?

**Comment by xiaoxq** [comment]
Per discussion, do merge commit.

**Comment by xiaoxq** [comment]
Per discussion, do merge-commit on this pull request to keep history.

**Pull request merged**